### PR TITLE
Feature/enhance create dataset

### LINKED ISSUE #27: Question on generating scores
Hello, 

I am able to run the example configuration given in the repo and the score generation to get the affinity scores for ICLR 2020. However, it is not clear to me what the config file should be for other years and conferences. For example, if I take the example configuration file and replace 2020 with 2019 it does not work. How can I figure out how to get the config for each conference for which there is data?

Thanks,

Tanner

### PR DESCRIPTION
**Pull request by carlosmondra**
It looks like a lot of changes, but I basically did 3 things that :

- I refactored the code in `create_dataset` so that it can be more general. It works with all the venues that I have tried, this potentially fixes issue #27.
- I also improved the way the `config.json` was loaded. Once it is loaded, the `ModelConfig`, which now iherits from `UserDict`, is instantiated.
- I replaced the `os` library with `libpath` as it seems to be more adequate from what I read and it is one of the issues that Michael created.

### CODE REVIEW

File: expertise/config/core.py [hunk 1]
```diff
@@ -1,53 +1,35 @@
-from collections import OrderedDict
+from collections import UserDict
 import json
 import random
-import os
+from pathlib import Path
```
Comment (mohituniyal) [inline expertise/config/core.py]: @carlosmondra It seems pathlib works with Py3.4 and above. I think we should we should put that somewhere in the documentation that we only support Py3.4 and above.

Comment (carlosmondra) [reply expertise/config/core.py]: Good point! I'll say that we support only python 3.6 and above though.

File: expertise/create_dataset.py [hunk 2]
```diff
@@ -6,27 +6,185 @@
...
 
 from collections import defaultdict, OrderedDict
 
-def search_with_retries(client, term, max_retries=5):
-    results = []
-    for iteration in range(max_retries):
-        try:
-            results = openreview_client.search_notes(
-                term=term, content='authors', group='all',source='forum')
+from config import ModelConfig
```
Comment (mohituniyal) [inline expertise/create_dataset.py]: This needs to be changed to `from .config import ModelConfig` to make this work.

**Comment by mohituniyal** [comment]
I noticed that having an experiment_dir in the config.json is inconsequential right now as we are always creating the dataset files in root dir.

**Comment by mohituniyal** [comment]
@carlosmondra The changes look good to me.

File: expertise/create_dataset.py (hunk 2 above)
Comment (carlosmondra) [reply expertise/create_dataset.py]: Fixed!

**Pull request merged**